test/e2e/e2e_test.go

```diff
@@ -0,0 +1,127 @@
+package e2e
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/xmudrii/etcdproxy-controller/pkg/apis/etcd/v1alpha1"
+	clientset "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// TestDeployEtcdStorage deploys EtcdStorage resource and then tests are all relevant resources created and running.
+func TestDeployEtcdStorage(t *testing.T) {
+	// Initialize clients.
+	cfg, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+	if err != nil {
+		t.Fatalf("unable to create kubernetes config from provided kubeconfig file: %v", err)
```

Comment:
nit: I generally just use t.Fatal(err).  You get a line number, so you can track it back easily enough.